src/lib/utils/evm.ts

```diff
@@ -104,15 +104,17 @@ export const getFees = async (args: GetFeesArgs): Promise<Fees> => {
     average: { chainSpecific: feeData },
   } = await adapter.getFeeData(getFeeDataInput)
 
+  const _supportsEIP1559 =
+    supportsEIP1559 ?? (supportsETH(wallet) && (await wallet.ethSupportsEIP1559()))
```

Comment:
might be useful to wrap this one with one that allows them to be optional, something like `getFeesWithHydrate` or something

Reply:
Yeah `wallet` was still here for backwards-compatibility but conceptually we should be able to pass everything we need from above instead of actually nullish coalescing here. Probably for a follow-up PR since this was just moved and isn't a new implementation